### PR TITLE
Remove redundant checking for code style in checklist

### PR DESCRIPTION
**Pull request by mdedetrich**
## Description

This checklist entry is redundant since we use scalafmt which actually enforces that code must be formatted even for it to pass the CI (especially after https://github.com/rallyhealth/scalacheck-ops/pull/57 is merged)

## Checklist:

- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by paul-a-kennedy-rally — APPROVED** [review]
LGTM

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -14,7 +14,6 @@ Fixes # (issue)
 
 ## Checklist:
 
-- [ ] My code follows the style guidelines of this project
```
Comment (jeffmay) [inline .github/pull_request_template.md:17]: While we auto-format the code with scalafmt, I think we may want to have a separate guidelines about "style" of code that is not related to formatting. That said, we don't have a style "guide" in this project, so I'm fine with removing this until we have one.
